lib/rubygems/ext/builder.rb

```diff
@@ -163,9 +161,8 @@ def build_extension extension, dest_path # :nodoc:
     # - @duckinator
     extension ||= '' # I wish I knew why this line existed
 
```

Comment:
My guess would be in case the extension is `nil`, otherwise the `File.dirname(extension)` below will fail. How that situation could arise, I'm not sure, but it must've happened at some point.

Reply:
Yup. I'm unsure what the problem is, and it's not commented, so more poking around will probably be needed.